src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -206,6 +205,30 @@ public String url() {
     /** True when first started, false when running after a restart. */
     private transient boolean firstTime;
 
+    private synchronized Object getLogCopyGuard() {
+        if (logCopyGuard == null) {
+            logCopyGuard = new Object();
+        }
+        return logCopyGuard;
+    }
+
+    /** Avoids creating new instances, analogous to {@link TaskListener#NULL} but as full StreamBuildListener. */
+    static final StreamBuildListener NULL_LISTENER = new StreamBuildListener(new NullStream());
+
+    /** Used internally to ensure listener has been initialized correctly. */
+    StreamBuildListener getListener() {
```

Comment:
Good catch. I believe there was a specific reason why it wasn't based on testing but... darned if I can remember the issue/exact chain of reasoning since it was rather complex.  

Give me a few, I'll either synchronize or provide a detailed comment to explain the rationale why not to.